src/ansible_navigator/image_manager/puller.py

```diff
@@ -22,22 +26,30 @@ class ImagePuller:
     # pylint: disable=too-many-instance-attributes
     """Image puller"""
 
-    def __init__(self, container_engine: str, image: str, pull_policy: str):
+    def __init__(
+        self,
+        container_engine: str,
+        image: str,
+        arguments: Union[Constants, List[str]],
+        pull_policy: str,
+    ):
         """Initialize the container image puller.
 
         :param container_engine: The name of the container engine
         :param image: The name of the image to pull
+        :param arguments: Additional arguments to be appended to the pull policy
         :param pull_policy: The current pull policy from the settings
         """
+        self._arguments: Union[Constants, List[str]] = arguments
```

Comment:
The self._arguments type can be set to list here itself instead of checking it later on?
Something like
```
self._arguments: List[str]
if not isinstance(arguments, list):
  self._arguments = []
else:
  self._arguments = arguments
```
```

Reply:
yeah, moving it